src/functional/ffront/parsing/linear_assign.py

```diff
@@ -0,0 +1,140 @@
+        return result[0]
+
+    def visit(self, node: ast.AST) -> Iterator[ast.AST]:
+        result = super().visit(node)
+        if isinstance(result, ast.AST):
+            yield result
+        else:
+            yield from result
+
+
+class SingleAssignTargetPass(NodeYielder):
+    """
+    Split multi target assignments.
+
+    Requires AST in SSA form (see ``SingleStaticAssignPass``) to yield correct results.
+
+    Example
+    -------
+    Usage:
+
+        import ast
+        from functional.ffront.parsing.parsing import get_ast_from_func
+
+        def foo():
+            a = b = 1
+            return a, b
+
+        print(ast.unparse(
+            SingleStaticAssignPass.mutate_ast(
+                get_ast_from_func(foo)
+            )
+        ))
+
+        # will print out:
+
+        def foo():
+            a = 1
+            b = 1
+
+    """
+
+    def visit_Assign(self, node: ast.Assign) -> Iterator[ast.Assign]:
+        for target in node.targets:
+            new_assign = copy.copy(node)
+            new_assign.targets = [target]
+            yield new_assign
+
+
+class UnpackedAssignPass(NodeYielder):
+    """
+    Explicitly unpack assignments.
+
+    Requires AST in SSA form and assumes only single target assigns, check the following passes.
+
+     * ``SingleStaticAssignPass``
+     * ``SingleAssignTargetPass``
+
+    Example
+    -------
+    Usage:
+
+        import ast
+        from functional.ffront.parsing.parsing import get_ast_from_func
+
+        def foo():
+            a0 = 1
+            b0 = 5
+            a1, b1 = b0, a0
+            return a1, b1
+
+        print(ast.unparse(
+            UnpackAssignPass.mutate_ast(
+                get_ast_from_func(foo)
+            )
+        ))
+
+        # will print out:
+
+        def foo():
+            a0 = 1
+            b0 = 5
+            a1 = (b0, a0)[0]
+            b1 = (b0, a0)[1]
+            return a1, b1
+
+        # which would not have been equivalent had the input AST not been in SSA form.
+    """
+
+    def visit_Assign(self, node: ast.Assign) -> Iterator[ast.Assign]:
+        if len(node.targets) != 1:
+            raise ValueError(
+                "AST contains multi target assigns. Please run `SingleAssignTargetPass` first."
+            )
+        if isinstance(target := node.targets[0], (ast.Tuple, ast.List)):
+            yield from self.unpack_assignment(node, targets=target.elts)
+        else:
+            yield node
+
+    def unpack_assignment(
+        self, node: ast.Assign, *, targets: list[ast.expr]  # targets passed here for typing
+    ) -> Iterator[ast.Assign]:
+        for index, subtarget in enumerate(targets):
+            new_assign = copy.copy(node)
+            new_assign.targets = [subtarget]
+            new_assign.value = ast.Subscript(
+                ctx=ast.Load(),  # <- for some reason this must be given for astpretty to work
```

Comment:
Well, `ctx` is not an optional attribute so it makes sense. In this case needs to be a `Load` context because it is used in the right hand side of the assignment, otherwise it would be an invalid AST... 
```suggestion
                ctx=ast.Load(),  # <- Load context in the right hand side of an assignment
```

Reply:
I will rephrase to emphasize more that `ctx` is not optional for Subscript, which strikes me as unusual.